projects/arcade-xyz/staking-queries.js

```diff
@@ -0,0 +1,82 @@
+const sdk = require("@defillama/sdk");
+const { sumTokensAndLPsSharedOwners } = require('../helper/unwrapLPs');
+const { getChainTransform } = require('../helper/portedTokens');
+
+const { CHAIN, STAKING_REWARDS, SINGLE_SIDED_STAKING, ARCD, ARCD_WETH_LP } = require('./constants');
+const { SINGLE_SIDED_STAKING_ABI, ARCD_ABI, LP_PAIR_ABI } = require("./abi.js");
+
+async function getTotalSupply(contractAddress, contractAbi, block) {
+  const totalSupply = (
+    await sdk.api.abi.call({
+      target: contractAddress,
+      abi: contractAbi.find((fn) => fn.name === "totalSupply"),
+      chain: CHAIN,
+      block: block,
+    })
+  ).output;
+
+  return totalSupply;
+}
+
+async function getBalanceOf(tokenAddress, holderAddress, abi, block) {
+  const balanceOf = (await sdk.api.abi.call({
+    target: tokenAddress,
+    abi: abi.find((fn) => fn.name === "balanceOf"),
+    params: [holderAddress],
+    block: block,
+  })).output;
+
+  return balanceOf;
+}
+
+async function getReserves(contractAddress, contractAbi, block) {
+  const getReserve = (await sdk.api.abi.call({
+    target: contractAddress,
+    abi: contractAbi.find((fn) => fn.name === "getReserves"),
+    block: block,
+  })).output;
+
+  return getReserve[1];
+}
+
+async function addToTVL(block, chainBlocks) {
+  const balances = {};
+
+  const transformAddress = await getChainTransform(CHAIN);
+
+  // handle LP token unwrapping and add the total supply
+  // for STAKING_REWARDS to balances
+  await sumTokensAndLPsSharedOwners(
+    balances,
+    [[ARCD_WETH_LP, true]],
+    [STAKING_REWARDS],
+    block,
+    CHAIN,
+    transformAddress
```

Comment:
here you need to pass something like:

```
    (tokenAddress) => {
      if (tokenAddress.toLowerCase()  === ARCD.toLowerCase()) {
        return `coingecko:arcade-protocol`
      }

      return tokenAddress
    }

```

Reply:
The [sumTokensAndLPsSharedOwners](https://github.com/arcadexyz/DefiLlama-Adapters/blob/74ecfc42ab50157808cde00889580c4a63e7ab3c/projects/helper/unwrapLPs.js#L110) function inherently does that for you.  It transforms the address by internally calling [getChainTransform](https://github.com/arcadexyz/DefiLlama-Adapters/blob/74ecfc42ab50157808cde00889580c4a63e7ab3c/projects/helper/portedTokens.js#L102).